examples/alert-absence-dedup/README.md

```diff
@@ -7,91 +7,119 @@ them is missing. If one or two timeseries are missing, you want exactly one
 alert. When there is a total outage you still want to get one alert, not 100
 alerts.
 
+The test app generates time series with a custom metric called
+task_latency_distribution with tags based on a partition label. Each
+partition generates its own time series.
+
 The example assumes that you are familiar with Stackdriver Monitoring and
 Alerting. It builds on the discussion in
 [Alerting policies in depth](https://cloud.google.com/monitoring/alerts/concepts-indepth).
 
-## Setup
+## Deploy the app
+
+The example code is based on the Go code in
+[Custom metrics with OpenCensus](https://cloud.google.com/monitoring/custom-metrics/open-census).
+
+[Download](https://golang.org/dl/) and install the latest version of Go.
 
-Create three GCE instances
+Build the test app
 
 ```shell
-ZONE=us-central1-c
-gcloud compute instances create test-instance-1 test-instance-2 test-instance-3 \
-  --zone=$ZONE \
-  --scopes=https://www.googleapis.com/auth/cloud-platform \
-  --boot-disk-size=20GB \
-  --machine-type=f1-micro
+go build
 ```
 
-Edit the file notification_channel.json, replacing the displayName and
-email_address. Create a notification channel with the command
+The instructions here as based on
+[Setting up authentication](https://cloud.google.com/monitoring/docs/reference/libraries#setting_up_authentication)
+for the Stackdriver client library. Note that if you run the test app on a
+Google Cloud Compute Engine instance, Google Kubernetes Engine, or App Engine
+you will not need to create a service account or download the credentials.
+
+First, set the project id in a shell variable
 
 ```shell
-gcloud alpha monitoring channels create \
-  --channel-content-from-file=notification_channel.json
+export GOOGLE_CLOUD_PROJECT=[your project]
 ```
 
-Note the name of the notification channel created for the next step.
+Create a service account:
 
 ```shell
-CHANNEL=[your notification channel]
+SA_NAME=stackdriver-service-account
+gcloud iam service-accounts create $SA_NAME
 ```
 
-Use of a [Stackdriver group](https://cloud.google.com/monitoring/groups/) is
-optional but it may help in identifying a specific set of
-resources to set up the alert for. Create a Stackdriver group with the three
-instances to be able to manage your metrics together. Find the ID of the group
-with the API Explorer using the
-[groups.list](https://cloud.google.com/monitoring/api/ref_v3/rest/v3/projects.groups/list)
-API. Note the ID of the group for use in the alert policy below.
+```shell
+SA_ID="$SA_NAME@$GOOGLE_CLOUD_PROJECT.iam.gserviceaccount.com"
+gcloud projects add-iam-policy-binding $GOOGLE_CLOUD_PROJECT \
+  --member "serviceAccount:$SA_ID" --role "roles/owner"
```

Comment:
This is granting a fairly broad role to the service account. I worry that such an example might be mistakenly emulated in a real case, being harmful to overall security for whoever follows this. Could you grant narrower access? How about "roles/monitoring.metricWriter" if all this is doing is reporting the time-series data?

Reply:
Changed to "roles/monitoring.metricWriter"